In [24]:

import handcalcs
from handcalcs.decorator import handcalc
from handcalcs import  render

handcalcs.set_option("math_environment_start", "aligned")
handcalcs.set_option("math_environment_end", "aligned")

handcalcs.set_option("display_precision", 2)
handcalcs.set_option("param_columns", 1)


handcalcs.set_option("latex_block_start", "$$")
handcalcs.set_option("latex_block_end", "$$")

handcalcs.set_option("preferred_string_formatter", "L")

import forallpeople as fp
fp.environment("structural")
m   = fp.m
mm  = fp.mm
kN  = fp.kN
MPa = fp.MPa
GPa = fp.GPa

## Design Load LRFD

In [25]:
%%render params
M_u = 860.9076 * kN * m
T_u = 90.743688 * kN
V_u = 203.728476 * kN
V_uc = 87.185112 * kN


<IPython.core.display.Latex object>

## Material Properties

In [26]:
%%render param
## Steel
E = (200 * GPa)

## Beam and column
F_ybc = (344.738 * MPa)
F_ubc = (448.1594 * MPa)

## Plate
F_yp = (344.738 * MPa)
F_up = (448.1594 * MPa)

## Stiffener
F_ys = (344.738 * MPa)
F_us = (448.1594 * MPa)

## Bolt
F_nt = (779.10788 * MPa)
F_nv = (468.84368 * MPa)

## Bolt pretension force
# For grade A490 bolt according to AISC Specification Table J3.1
T_b = (453.71844 * kN)

<IPython.core.display.Latex object>

## Geometry Properties

In [27]:
%%render param

## Beam
d_b = 612.14 * mm # web height
b_bf = 229.108 * mm # flange width
t_bw = 11.938 * mm # web thick
t_bf = 19.558 * mm # flange thick

## Column
d_c = 360.68 * mm # web height
b_cf = 370.84 * mm # flange width
t_cw = 12.319 * mm # web thick
t_cf = 19.812 * mm # flange thick
k_des = 35.052 * mm # fillet distance used in production

## Plate
b_p = 228.6 * mm # width
t_p = 22.225 * mm # thickness

## Bolt 
d_bolt = 31.75 * mm # diameter
d_hole = d_bolt + 3.175 * mm # hole diameter

## Bolt configuration
g = 146.05 * mm # gauge
p_ext = 114.3 * mm # top plate to beam
d_e = 44.45 * mm # edge distance
p_fo = 69.85 * mm # flange outer bolt
p_fi = 50.8 * mm # flange inner bolt
d_p = 841.375 * mm # bolt group height

## Stiffener
t_s = 12.7 * mm # thickness
L_st = 203.2 * mm # length
h_st = 114.3 * mm # height

<IPython.core.display.Latex object>

#### Calculation

In [28]:
%%render intertext
## 1. The distance from the centerline of the compression flange to the bolt lines

## 1.1 Calculate distance
h_1 = d_b - (t_bf / 2) + p_fo

h_2 = d_b - (3 * t_bf / 2) - p_fi

## 1.2 Check base plate effective width
b_p_limit = b_bf + max(t_p, 25.4 * mm)
if b_p <= b_p_limit : b_p_eff = b_p
elif b_p > b_p_limit : b_p_eff = b_p_limit

## 1.3 Check bolt gage
b_bf_limit = b_bf
if b_bf >= b_bf_limit : b_bf = b_bf_limit # OK
elif b_bf < b_bf_limit : b_bf = b_bf_limit # NOT OK


## 1. The distance from the centerline of the compression flange to the bolt lines

### 1.1 Calculate distance

$$
\begin{aligned}
h_{1} &= d_{b} - \left( \frac{ t_{bf} }{ 2 } \right) + p_{fo} \\&= 612.14\ \mathrm{mm} - \left( \frac{ 19.56\ \mathrm{mm} }{ 2 } \right) + 69.85\ \mathrm{mm} \\&= 672.21\ \mathrm{mm}  

h_{2} &= d_{b} - \left( 3 \cdot \frac{ t_{bf} }{ 2 } \right) - p_{fi} \\&= 612.14\ \mathrm{mm} - \left( 3 \cdot \frac{ 19.56\ \mathrm{mm} }{ 2 } \right) - 50.80\ \mathrm{mm} \\&= 532.00\ \mathrm{mm}  
\end{aligned}
$$

### 1.2 Check base plate effective width

$$
\begin{aligned}
b_{p_{limit}} &= b_{bf} + \operatorname{max} \left( t_{p} ,\  25.4 \cdot \mathrm{mm} \right) \\&= 229.11\ \mathrm{mm} + \operatorname{max} \left( 22.23\ \mathrm{mm} ,\  25.4 \cdot 1.00\ \mathrm{mm} \right) \\&= 254.51\ \mathrm{mm}  

&\text{Since, } b_{p} \leq b_{p_{limit}} \rightarrow \left( 228.60\ \mathrm{mm} \leq 254.51\ \mathrm{mm} \right) :   \\[10pt]
b_{p_{eff}} &= 228.60\ \mathrm{mm} \; 
\end{aligned}
$$

### 1.3 Check bolt gage

$$
\begin{aligned}
b_{bf_{limit}} &= 229.11\ \mathrm{mm} \; 

&\text{Since, } b_{bf} \geq b_{bf_{limit}} \rightarrow \left( 229.11\ \mathrm{mm} \geq 229.11\ \mathrm{mm} \right) : \; \;\textrm{(OK)} \\[10pt]
b_{bf} &= 229.11\ \mathrm{mm} \; 
\end{aligned}
$$

In [29]:
%%tex intertext
## 1. The distance from the centerline of the compression flange to the bolt lines

## 1.1 Calculate distance
h_1 = d_b - (t_bf / 2) + p_fo

h_2 = d_b - (3 * t_bf / 2) - p_fi

## 1.2 Check base plate effective width
b_p_limit = b_bf + max(t_p, 25.4 * mm)
if b_p <= b_p_limit : b_p_eff = b_p
elif b_p > b_p_limit : b_p_eff = b_p_limit

## 1.3 Check bolt gage
b_bf_limit = b_bf
if b_bf >= b_bf_limit : b_bf = b_bf_limit # OK
elif b_bf < b_bf_limit : b_bf = b_bf_limit # NOT OK


## 1. The distance from the centerline of the compression flange to the bolt lines

### 1.1 Calculate distance

$$
\begin{aligned}
h_{1} &= d_{b} - \left( \frac{ t_{bf} }{ 2 } \right) + p_{fo} \\&= 612.14\ \mathrm{mm} - \left( \frac{ 19.56\ \mathrm{mm} }{ 2 } \right) + 69.85\ \mathrm{mm} \\&= 672.21\ \mathrm{mm}  

h_{2} &= d_{b} - \left( 3 \cdot \frac{ t_{bf} }{ 2 } \right) - p_{fi} \\&= 612.14\ \mathrm{mm} - \left( 3 \cdot \frac{ 19.56\ \mathrm{mm} }{ 2 } \right) - 50.80\ \mathrm{mm} \\&= 532.00\ \mathrm{mm}  
\end{aligned}
$$

### 1.2 Check base plate effective width

$$
\begin{aligned}
b_{p_{limit}} &= b_{bf} + \operatorname{max} \left( t_{p} ,\  25.4 \cdot \mathrm{mm} \right) \\&= 229.11\ \mathrm{mm} + \operatorname{max} \left( 22.23\ \mathrm{mm} ,\  25.4 \cdot 1.00\ \mathrm{mm} \right) \\&= 254.51\ \mathrm{mm}  

&\text{Since, } b_{p} \leq b_{p_{limit}} \rightarrow \left( 228.60\ \mathrm{mm} \leq 254.51\ \mathrm{mm} \right) :   \\[10pt]
b_{p_{eff}} &= 228.60\ \mathrm{mm} \; 
\end

$$
\begin{array}{l}
% --- Header ---
\hspace{-5em}\text{\textbf{\Large 1. The distance from the centerline of the compression flange to the bolt lines}} \\[8pt]

% --- Subheader ---
\hspace{-5em}\text{\textbf{\large 1.1 Calculate distance}} \\[10pt]

% --- Description paragraph ---
\text{This section evaluates the effective distance from the compression flange centerline}
\\
\text{to the bolt lines based on the beam geometry and flange thickness.}
\\[12pt]

% --- Explanation / logic ---
\text{Since, } b_p \le b_{p_{\text{limit}}}
\;\rightarrow\;
(228.60\ \mathrm{mm} \le 254.51\ \mathrm{mm}) \\[12pt]

% --- Equation ---
\begin{alignedat}{2}
b_{p_{\text{eff}}} &{}= 228.60\ \mathrm{mm}
\end{alignedat}
\end{array}
$$
